src/PrestaShopBundle/Controller/Admin/Sell/Order/InvoicesController.php

```diff
@@ -0,0 +1,223 @@
+     */
+    public function indexAction(Request $request)
+    {
+        $legacyController = $request->attributes->get('_legacy_controller');
+
+        $byDateForm = $this->get('prestashop.admin.order.invoices.by_date.form_handler')->getForm();
+        $byStatusForm = $this->get('prestashop.admin.order.invoices.by_status.form_handler')->getForm();
+        $optionsForm = $this->get('prestashop.admin.order.invoices.options.form_handler')->getForm();
+
+        return [
+            'layoutTitle' => $this->trans('Invoices', 'Admin.Navigation.Menu'),
+            'requireAddonsSearch' => true,
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($legacyController),
+            'byDateForm' => $byDateForm->createView(),
+            'byStatusForm' => $byStatusForm->createView(),
+            'optionsForm' => $optionsForm->createView(),
+        ];
+    }
+
+    /**
+     * Action that generates invoices PDF by date interval
+     *
+     * @param Request $request
+
+     * @return RedirectResponse
+     */
+    public function generatePdfByDateAction(Request $request)
+    {
+        $legacyController = $request->attributes->get('_legacy_controller');
+
+        if (!in_array(
+            $this->authorizationLevel($legacyController),
+            [
+                PageVoter::LEVEL_UPDATE,
+                PageVoter::LEVEL_CREATE,
+                PageVoter::LEVEL_DELETE,
+            ]
+        )) {
+            $this->addFlash(
+                'error',
+                $this->trans('You do not have permission to edit this', 'Admin.Notifications.Error')
+            );
+
+            return $this->redirectToRoute('admin_order_invoices');
+        }
+
+        $formHandler = $this->get('prestashop.admin.order.invoices.by_date.form_handler');
+
+        $form = $formHandler->getForm();
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $data = $form->getData();
+
+            if ($errors = $formHandler->save($data)) {
+                $this->flashErrors($errors);
+            } else {
+                $invoiceDataProvider = $this->get('prestashop.adapter.data_provider.order_invoice');
+
+                // Get invoices by submitted date interval
+                $invoiceCollection = $invoiceDataProvider->getByDateInterval(
+                    $data['generate_by_date']['date_from'],
+                    $data['generate_by_date']['date_to']
+                );
+
+                if (empty($invoiceCollection)) {
+                    $this->addFlash(
+                        'error',
+                        $this->trans('No invoice was found.', 'Admin.Orderscustomers.Notification')
+                    );
+                } else {
+                    // Generate PDF out of found invoices
+                    $invoiceGenerator = $this->get('prestashop.adapter.pdf_generator.invoice');
+                    $invoiceGenerator->generateInvoicesPDF($invoiceCollection);
```

Comment:
could be worth to make this array a typed Collection, wdyt @sarjon ?

Reply:
in all cases there is too much business logic here: why it's not done when doing `$formHandler->save($data)`?

If the abstract class have too many constraints, you can rely on the interface :)